### PR TITLE
BHBC-1828: Display Spatial Components on Map

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to Jira tickets

https://quartech.atlassian.net/browse/BHBC-1828

### API
Add spatial search endpoint.
- Will be responsible for fetching all types of spatial components

### APP
Update MapPage and related components
- Marker clusters now render as a layer (can be toggled on/off with built in leaflet UI control)
- Etc

Update useDataLoader hook and tests 
- Supports calling refresh with different parameters
- Etc

Update useDataLoaderError hook
- Has better defaults for the error dialog, so that in most cases you only need to supply the title/text now.

## Testing
Zip that contains a SQL file (to run in dbeaver or similar) with some commands for inserting a sample submission record, and sample spatial components (2 boundaries, and 9 occurrence points). 
[spatial_component_test.zip](https://github.com/bcgov/biohubbc-platform/files/9041882/spatial_component_test.zip)

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: app/src/components/map/OccurrenceFeaturePopup.tsx [hunk 1]
```diff
@@ -1,76 +1,19 @@
...
-                  );
-                })}
-              </Box>
-            </div>
-          );
-        })}
-      </div>
-    </Popup>
+    <div>
+      <div>{JSON.stringify(properties)}</div>
```
Comment (NickPhura) [inline app/src/components/map/OccurrenceFeaturePopup.tsx:16]: To be enhanced in a later ticket.

File: app/src/components/map/BoundaryFeaturePopup.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export type BoundaryFeatureProperties = {
+  type: SPATIAL_COMPONENT_TYPE.BOUNDARY;
+};
+
+export const BoundaryFeaturePopup: React.FC<{ properties: BoundaryFeatureProperties }> = (props) => {
+  const { properties } = props;
+
+  return (
+    <div>
+      <div>{JSON.stringify(properties)}</div>
```
Comment (NickPhura) [inline app/src/components/map/BoundaryFeaturePopup.tsx:16]: To be enhanced in a later ticket.

**Review by curtisupshall — DISMISSED** [review]
Nice additions to `useDataLoader` 😎

**Review by curtisupshall — DISMISSED** [review]
👌

**Comment by NickPhura** [comment]
@curtisupshall sorry, 1 final commit :)  That should be all though.

**Review by curtisupshall — APPROVED** [review]
+1 for more explicit type names 😃

**Pull request merged**